packages/cli-internal/src/commands/push.ts

```diff
@@ -411,10 +411,17 @@ export function getOptions(
       tags.push('authentication:test') //valid values here can eventually be test, no-test if needed
     }
 
+    const defaultValues = {
+      number: 0,
+      boolean: false,
+      string: '',
+      password: ''
+    }
+
     options[fieldKey] = {
       ...existing,
       tags,
-      default: schema.default ?? '',
+      default: schema.default ?? defaultValues[schema.type],
```

Comment:
This should ensure that we always have some default value in case `defaultValues[schema.type]` is undefined.

```suggestion
      default: schema.default ?? (defaultValues[schema.type] || ''),
```

Reply:
I dont believe it can be undefined given typescript would throw of the schema type didnt match one of those values

Reply:
I see, the `GlobalSetting` type right now, where type can only be one of these four options - you are right. Thanks for clarifying!